File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-11.go
func findMin(nums []int) int {
    n := len(nums)
    l := 0
    r := n - 1
    res := nums[l]
   
    for l <= r {
        if nums[l] < nums[r] {
            if nums[l] < res {
                res = nums[l]
            }
            break;
        }

        m := (l + r) / 2
       
        if nums[m] < res {
            res = nums[m]
        }

        if nums[m] > nums[r]{
            l = m + 1
        }else {
            r = m - 1
        }
    }
    return res
}

File: Data Structures & Algorithms/two-integer-sum/submission-11.cs
public class Solution {
    public int[] TwoSum(int[] nums, int target) {

        var dic = new Dictionary<int,int>();

        for(var i= 0; i < nums.Length ; i++) {
            var res = target - nums[i];

            if(dic.ContainsKey(res)){
                return new int[] {dic[res] , i};
            }
            else 
                dic[nums[i]] = i;
        }
        return new int[] {};
    }
}
